statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,698 @@
+#                                         self.fit_kwds,
+#                                         **self.estimation_kwds)
+#        return results
+
+
+    def fit_distributed(self, parallel_method="sequential"):
+        """Performs the distributed estimation using the corresponding
+        DistributedModel
+
+        Parameters
+        ----------
+        parallel_method : str
+            type of distributed estimation to be used, currently
+            "sequential", "joblib" and "dask" are supported.
+
+        Returns
+        -------
+        join_method result.  For the default, _join_debiased, it returns a
+        p length array.
+        """
+
+        if parallel_method == "sequential":
+            return self.fit_dist_sequential()
+
+        elif parallel_method == "joblib":
+            try:
+                return self.fit_dist_joblib()
+            except ImportError:
+                raise ValueError("cannot import joblib")
+
+        elif parallel_method == "dask":
+            try:
+                return self.fit_dist_dask()
+            except ImportError:
+                raise ValueError("cannot import dask")
+
+        else:
+            raise ValueError("parallel_method: %s is currently not supported"
+                             % parallel_method)
+
+
+    def fit_dist_sequential(self):
+        """Sequentially performs the distributed estimation using
+        the corresponding DistributedModel
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        join_method result.  For the default, _join_debiased, it returns a
+        p length array.
+        """
+
+        results_l = []
+
+        for pnum, (endog, exog) in enumerate(self.data_generator):
+
+            results = _helper_fit_partition(self, pnum, endog, exog)
+            results_l.append(results)
+
+        return self.join_method(results_l, self.partitions,
+                                **self.join_kwds)
+
+
+    def fit_dist_joblib(self):
+        """Performs the distributed estimation in parallel using joblib
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        join_method result.  For the default, _join_debiased, it returns a
+        p length array.
+        """
+
+#        try:
+#            from joblib import Parallel, delayed
+#        except ImportError:
+#            from sklearn.externals.joblib import Parallel, delayed
+
+        from statsmodels.tools.parallel import parallel_func
+
+        par, f, n_jobs = parallel_func(_helper_fit_partition, self.partitions)
+
+#        par = Parallel(n_jobs=self.partitions)
+#        f = delayed(_helper_fit_partition)
+        results_l = par(f(self, pnum, endog, exog)
+                        for pnum, (endog, exog)
+                        in enumerate(self.data_generator))
+        return self.join_method(results_l, self.partitions,
+                                **self.join_kwds)
+
+
+    def fit_dist_dask(self):
```

Comment:
I think we can avoid having this here.  Instead, now (or later) we can extend `statsmodels.tools.parallel` to use dask (if it is available).